.github/workflows/new-benchmark.yml

```diff
@@ -0,0 +1,56 @@
+name: New Benchmark
+
+on:
+  push:
+    branches:
+      - master
+
+permissions:
+  # Deployments permission to deploy GitHub pages website
+  deployments: write
+  # Contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  new-benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.3.0
+
+      - name: Run benchmark
```

Comment:
I think it's worthwhile doing this in its own job so it alone can run on the benchmarking machine, and it's isolated.

Reply:
Also then we can run this as a matrix job if we want to do multiple benchmarks in the future.